File: address-validation-canada-2-python-examples/validate_canadian_address_v2_soap_sdk_example.py
import sys
import os

sys.path.insert(0, os.path.abspath("../address-validation-canada-2-python/SOAP"))

from validate_canadian_address_v2_soap import ValidateCanadianAddressV2Soap


def validate_canadian_address_v2_soap_sdk_go(is_live: bool, license_key: str) -> None:
    print("\r\n--------------------------------------------")
    print("AVCA2 - ValidateCanadianAddressV2 - SOAP SDK")
    print("--------------------------------------------")

    address = "50 Coach Hill Dr"
    address2 = ""
    municipality = "Kitchener"
    province = "Ont"
    postal_code = "N2E 1P4"
    language = "English"
    timeout_seconds = 15

    print("\r\n* Input *\r\n")
    print(f"Address        : {address}")
    print(f"Address2       : {address2}")
    print(f"Municipality   : {municipality}")
    print(f"Province       : {province}")
    print(f"Postal Code    : {postal_code}")
    print(f"Language       : {language}")
    print(f"License Key    : {license_key}")
    print(f"Is Live        : {is_live}")
    print(f"Timeout Seconds: {timeout_seconds}")

    try:
        service = ValidateCanadianAddressV2Soap(license_key, is_live, timeout_seconds * 1000)
        response = service.validate_canadian_address_v2(address, address2, municipality, province, postal_code, language)

        if not hasattr(response, 'Error') or not response.Error:
            print("\r\n* Address Info *\r\n")
            if hasattr(response, 'CanadianAddressInfoV2') and response.CanadianAddressInfoV2:
                address_info = response.CanadianAddressInfoV2
                print(f"Address              : {getattr(address_info, 'Address', None)}")
                print(f"Address2             : {getattr(address_info, 'Address2', None)}")
                print(f"Municipality         : {getattr(address_info, 'Municipality', None)}")
                print(f"Province             : {getattr(address_info, 'Province', None)}")
                print(f"Postal Code          : {getattr(address_info, 'PostalCode', None)}")
                print(f"Time Zone            : {getattr(address_info, 'TimeZone', None)}")
                print(f"Address Number       : {getattr(address_info, 'AddressNumberFragment', None)}")
                print(f"Street Name          : {getattr(address_info, 'StreetNameFragment', None)}")
                print(f"Street Type          : {getattr(address_info, 'StreetTypeFragment', None)}")
                print(f"Directional Code     : {getattr(address_info, 'DirectionalCodeFragment', None)}")
                print(f"Unit Type            : {getattr(address_info, 'UnitTypeFragment', None)}")
                print(f"Unit Number          : {getattr(address_info, 'UnitNumberFragment', None)}")
                print(f"Is PO Box            : {getattr(address_info, 'IsPOBox', None)}")
                print(f"Box Number           : {getattr(address_info, 'BoxNumberFragment', None)}")
                print(f"Station Info         : {getattr(address_info, 'StationInfo', None)}")
                print(f"Delivery Mode        : {getattr(address_info, 'DeliveryMode', None)}")
                print(f"Delivery Installation: {getattr(address_info, 'DeliveryInstallation', None)}")
                print(f"Corrections          : {getattr(address_info, 'Corrections', None)}")
                print(f"Corrections Desc     : {getattr(address_info, 'CorrectionsDescriptions', None)}")
            else:
                print("No CanadianAddressInfoV2 found.")
        else:
            print("No address info found.")

        if hasattr(response, 'Error') and response.Error:
            print("\r\n* Error *\r\n")
            print(f"Error Type      : {getattr(response.Error, 'Type', None)}")
            print(f"Error Type Code : {getattr(response.Error, 'TypeCode', None)}")
            print(f"Error Desc      : {getattr(response.Error, 'Desc', None)}")
            print(f"Error Desc Code : {getattr(response.Error, 'DescCode', None)}")

    except Exception as e:
        print("\r\n* Error *\r\n")
        print(f"Exception occurred: {str(e)}")

File: address-validation-canada-2-nodejs-examples/validate_canadian_address_v2_soap_sdk_example.js
import { ValidateCanadianAddressV2Soap } from '../address-validation-canada-2-nodejs/SOAP/validate_canadian_address_v2_soap.js';

async function ValidateCanadianAddressV2SoapGo(licenseKey, isLive) {
    console.log("\n--------------------------------------------");
    console.log("AVCA2 - ValidateCanadianAddressV2 - SOAP SDK");
    console.log("--------------------------------------------");

    const address = "50 Coach Hill Dr";
    const address2 = "";
    const municipality = "Kitchener";
    const province = "Ont";
    const postalCode = "N2E 1P4";
    const language = "English";
    const timeoutSeconds = 15;

    console.log("\n* Input *\n");
    console.log(`Address        : ${address}`);
    console.log(`Address2       : ${address2}`);
    console.log(`Municipality   : ${municipality}`);
    console.log(`Province       : ${province}`);
    console.log(`Postal Code    : ${postalCode}`);
    console.log(`Language       : ${language}`);
    console.log(`License Key    : ${licenseKey}`);
    console.log(`Is Live        : ${isLive}`);
    console.log(`Timeout Seconds: ${timeoutSeconds}`);

    try {
        const avca2 = new ValidateCanadianAddressV2Soap(
            address,
            address2,
            municipality,
            province,
            postalCode,
            language,
            licenseKey,
            isLive,
            timeoutSeconds
        );
        const response = await avca2.invokeAsync();

        if (response.Error) {
            console.log("\n* Error *\n");
            console.log(`Error Type      : ${response.Error.Type}`);
            console.log(`Error Type Code : ${response.Error.TypeCode}`);
            console.log(`Error Desc      : ${response.Error.Desc}`);
            console.log(`Error Desc Code : ${response.Error.DescCode}`);
            return;
        }

        console.log("\n* Address Info *\n");
        if (response) {
            if (response.CanadianAddressInfoV2) {
                console.log(`Address              : ${response.CanadianAddressInfoV2.Address}`);
                console.log(`Address2             : ${response.CanadianAddressInfoV2.Address2}`);
                console.log(`Municipality         : ${response.CanadianAddressInfoV2.Municipality}`);
                console.log(`Province             : ${response.CanadianAddressInfoV2.Province}`);
                console.log(`Postal Code          : ${response.CanadianAddressInfoV2.PostalCode}`);
                console.log(`Time Zone            : ${response.CanadianAddressInfoV2.TimeZone}`);
                console.log(`Address Number       : ${response.CanadianAddressInfoV2.AddressNumberFragment}`);
                console.log(`Street Name          : ${response.CanadianAddressInfoV2.StreetNameFragment}`);
                console.log(`Street Type          : ${response.CanadianAddressInfoV2.StreetTypeFragment}`);
                console.log(`Directional Code     : ${response.CanadianAddressInfoV2.DirectionalCodeFragment}`);
                console.log(`Unit Type            : ${response.CanadianAddressInfoV2.UnitTypeFragment}`);
                console.log(`Unit Number          : ${response.CanadianAddressInfoV2.UnitNumberFragment}`);
                console.log(`Is PO Box            : ${response.CanadianAddressInfoV2.IsPOBox}`);
                console.log(`Box Number           : ${response.CanadianAddressInfoV2.BoxNumberFragment}`);
                console.log(`Station Info         : ${response.CanadianAddressInfoV2.StationInfo}`);
                console.log(`Delivery Mode        : ${response.CanadianAddressInfoV2.DeliveryMode}`);
                console.log(`Delivery Installation: ${response.CanadianAddressInfoV2.DeliveryInstallation}`);
                console.log(`Corrections          : ${response.CanadianAddressInfoV2.Corrections}`);
                console.log(`Corrections Desc     : ${response.CanadianAddressInfoV2.CorrectionsDescriptions}`);
            } else {
                console.log("No CanadianAddressInfoV2 found.");
            }
        } else {
            console.log("No address info found.");
        }
    } catch (e) {
        console.log("\n* Error *\n");
        console.log(`Error Message: ${e.message}`);
    }
}

export { ValidateCanadianAddressV2SoapGo };